test-data/unit/check-inference.test

```diff
@@ -1940,6 +1940,12 @@ class A:
 [out]
 main:4: error: "None" has no attribute "__iter__" (not iterable)
 
+[case testPartialTypeErrorSpecialCase4]
+# This used to crash.
+arr = []
+arr.append(arr.append(1))
+[out]
+main:3: error: "List[int]" has no attribute "append"
```

Comment:
It's better than a crash (thanks for filing this PR!), but this error still doesn't make sense.

Reply:
@JelleZijlstra  You are right, that is because I forgot to add fixtures to a test. Fixed, please, review again.

Reply:
Thanks! I guess when there's something weird going on in a mypy test there's a 90% chance it's due to a fixture.